src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -10,32 +13,48 @@
      * ставка за 1 час. Метод должен вернуть отчёт за период, который передали в метод
      * (обе даты включительно) составленный по следующей форме: Отчёт за период
      * #дата_1# - #дата_2# Имя сотрудника - сумма заработанных средств за этот период
-     *
      * Пример ввода: date from = 01.04.2019 date to = 30.04.2019
-     *
      * names:
      * Сергей
      * Андрей
      * София
-     *
      * data:
      * 26.04.2019 Сергей 60 50
      * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50
-     *
+     * 27.04.2019 Сергей 7 100
+     * 28.04.2019 София 9 100
+     * 28.04.2019 Сергей 11 50
+     * 28.04.2019 Андрей 3 200
+     * 29.04.2019 Сергей 7 100
+     * 29.04.2019 София 9 100
+     * 30.04.2019 Сергей 11 50
      * Пример вывода:
      * Отчёт за период 01.04.2019  - 30.04.2019
      * Сергей - 1550
      * Андрей - 600
      * София - 900
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+        if (localDateTo.isBefore(LocalDate.parse(dateFrom, formatter))) {
+            return null;
+        }
+        String list = "Отчёт за период " + dateFrom + " - " + dateTo + "\n";
+        for (String name : names) {
+            int salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                String[] stringData = data[i].split(" ");
+                LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
```

Comment:
move this line before first `if` and try to use this variable in the if block:
if (localDateTo.isBefore(localDateFrom)) {
            return null;
        }

Reply:
fixed, if I understood correctly